src/spec/operations.ts

```diff
@@ -67,11 +67,29 @@ function getOperationResponses(op: any): ApiOperationResponse[] {
 }
 
 function getOperationSecurity(op: any): ApiOperationSecurity[] {
-  if (!op.security || !op.security.length) return
+  if (!op.security) return;
 
-  return op.security.map(def => {
-    const id = Object.keys(def)[0]
-    const scopes = def[id].length ? def[id] : undefined
-    return { id, scopes }
-  })
+  if (op.security.length) {
+    return op.security.map(def => {
+      const id = Object.keys(def)[0];
+      const scopes = def[id].length ? def[id] : undefined;
+      return { id: id, scopes: scopes };
+    });
+  } else if (Object.keys(op.security).length) {
+    return [{ id: Object.keys(op.security)[0] }];
+  }
+}
+
+function getApiSecurity(spec: any): ApiOperationSecurity[] {
```

Comment:
This function looks the similar to the one above. Could we have one function that handles `op.security` and `spec.security` as they're the same shape.

I'd be happy with something like `op.security = getOperationSecurity(op, spec)` which handled the fallback internally and then run the same formatting code given the chosen security array.

Reply:
Agreed, will update with these chages